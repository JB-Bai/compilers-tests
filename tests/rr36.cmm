float vMdh8BfspiP(struct dnXtx1{
	struct{
	}
	tsoKIRA9Y = 0xC2d, RDF41dT = iYKTZMh.acELoukjqYl;
	struct CTRBH{
	}
	MaM7xM7DT, Orul55UAXtD = ! - 21.92 + yjhP2T0e(4630) / 0163614 || 619.566[oqjSfH(- - 00126[! rYNuwdBh] - 0xf859 =(gu58z0()) / vo1NbbZF != 0116205 - 0716 <= 5.6098E-6 || sWq8gXify() || Jd8mJR()) >= NS13U96gk() || futEz().iw7jwOiJyheP];
}
PaiyNf5dt){
	fd0mIDTgBq() || ! 2648 / - YNMncjU() * FwPWTmKKA[0552] > 060437 > brHgVtB[- IwXhgEsJI = 729.797E+3];
}
float;
float;
float IHWH8Sa3Q;

