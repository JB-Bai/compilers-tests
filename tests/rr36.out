Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      FunDec (1)
        ID: vMdh8BfspiP
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              StructSpecifier (1)
                STRUCT
                OptTag (1)
                  ID: dnXtx1
                LC
                DefList (2)
                  Def (2)
                    Specifier (2)
                      StructSpecifier (2)
                        STRUCT
                        LC
                        RC
                    DecList (4)
                      Dec (4)
                        VarDec (4)
                          ID: tsoKIRA9Y
                        ASSIGNOP
                        Exp (4)
                          INT: 3117
                      COMMA
                      DecList (4)
                        Dec (4)
                          VarDec (4)
                            ID: RDF41dT
                          ASSIGNOP
                          Exp (4)
                            Exp (4)
                              ID: iYKTZMh
                            DOT
                            ID: acELoukjqYl
                    SEMI
                  DefList (5)
                    Def (5)
                      Specifier (5)
                        StructSpecifier (5)
                          STRUCT
                          OptTag (5)
                            ID: CTRBH
                          LC
                          RC
                      DecList (7)
                        Dec (7)
                          VarDec (7)
                            ID: MaM7xM7DT
                        COMMA
                        DecList (7)
                          Dec (7)
                            VarDec (7)
                              ID: Orul55UAXtD
                            ASSIGNOP
                            Exp (7)
                              Exp (7)
                                Exp (7)
                                  NOT
                                  Exp (7)
                                    MINUS
                                    Exp (7)
                                      FLOAT: 21.920000
                                PLUS
                                Exp (7)
                                  Exp (7)
                                    ID: yjhP2T0e
                                    LP
                                    Args (7)
                                      Exp (7)
                                        INT: 4630
                                    RP
                                  DIV
                                  Exp (7)
                                    INT: 59276
                              OR
                              Exp (7)
                                Exp (7)
                                  FLOAT: 619.565979
                                LB
                                Exp (7)
                                  Exp (7)
                                    Exp (7)
                                      ID: oqjSfH
                                      LP
                                      Args (7)
                                        Exp (7)
                                          Exp (7)
                                            Exp (7)
                                              MINUS
                                              Exp (7)
                                                MINUS
                                                Exp (7)
                                                  Exp (7)
                                                    INT: 86
                                                  LB
                                                  Exp (7)
                                                    NOT
                                                    Exp (7)
                                                      ID: rYNuwdBh
                                                  RB
                                            MINUS
                                            Exp (7)
                                              INT: 63577
                                          ASSIGNOP
                                          Exp (7)
                                            Exp (7)
                                              Exp (7)
                                                Exp (7)
                                                  Exp (7)
                                                    Exp (7)
                                                      LP
                                                      Exp (7)
                                                        ID: gu58z0
                                                        LP
                                                        RP
                                                      RP
                                                    DIV
                                                    Exp (7)
                                                      ID: vo1NbbZF
                                                  RELOP
                                                  Exp (7)
                                                    Exp (7)
                                                      INT: 40069
                                                    MINUS
                                                    Exp (7)
                                                      INT: 462
                                                RELOP
                                                Exp (7)
                                                  FLOAT: 0.000006
                                              OR
                                              Exp (7)
                                                ID: sWq8gXify
                                                LP
                                                RP
                                            OR
                                            Exp (7)
                                              ID: Jd8mJR
                                              LP
                                              RP
                                      RP
                                    RELOP
                                    Exp (7)
                                      ID: NS13U96gk
                                      LP
                                      RP
                                  OR
                                  Exp (7)
                                    Exp (7)
                                      ID: futEz
                                      LP
                                      RP
                                    DOT
                                    ID: iw7jwOiJyheP
                                RB
                      SEMI
                RC
            VarDec (9)
              ID: PaiyNf5dt
        RP
      CompSt (9)
        LC
        StmtList (10)
          Stmt (10)
            Exp (10)
              Exp (10)
                ID: fd0mIDTgBq
                LP
                RP
              OR
              Exp (10)
                Exp (10)
                  Exp (10)
                    Exp (10)
                      Exp (10)
                        NOT
                        Exp (10)
                          INT: 2648
                      DIV
                      Exp (10)
                        MINUS
                        Exp (10)
                          ID: YNMncjU
                          LP
                          RP
                    STAR
                    Exp (10)
                      Exp (10)
                        ID: FwPWTmKKA
                      LB
                      Exp (10)
                        INT: 362
                      RB
                  RELOP
                  Exp (10)
                    INT: 24863
                RELOP
                Exp (10)
                  Exp (10)
                    ID: brHgVtB
                  LB
                  Exp (10)
                    Exp (10)
                      MINUS
                      Exp (10)
                        ID: IwXhgEsJI
                    ASSIGNOP
                    Exp (10)
                      FLOAT: 729797.000000
                  RB
            SEMI
        RC
    ExtDefList (12)
      ExtDef (12)
        Specifier (12)
          TYPE: float
        SEMI
      ExtDefList (13)
        ExtDef (13)
          Specifier (13)
            TYPE: float
          SEMI
        ExtDefList (14)
          ExtDef (14)
            Specifier (14)
              TYPE: float
            ExtDecList (14)
              VarDec (14)
                ID: IHWH8Sa3Q
            SEMI
